cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -1402,15 +1402,19 @@ macro(build_rapidjson)
   add_dependencies(rapidjson rapidjson_ep)
 endmacro()
 
-# TODO: Check for 1.1.0+
 if(ARROW_WITH_RAPIDJSON)
+  set(ARROW_RAPIDJSON_REQUIRED_VERSION "1.1.0")
   if(RapidJSON_SOURCE STREQUAL "AUTO")
     # Fedora packages place the package information at the wrong location.
     # https://bugzilla.redhat.com/show_bug.cgi?id=1680400
-    find_package(RapidJSON QUIET HINTS "${CMAKE_ROOT}")
+    find_package(RapidJSON
+                 ${ARROW_RAPIDJSON_REQUIRED_VERSION}
+                 QUIET
+                 HINTS
+                 "${CMAKE_ROOT}")
     if(NOT RapidJSON_FOUND)
       # Ubuntu / Debian don't package the CMake config
-      find_package(RapidJSONAlt)
+      find_package(RapidJSONAlt ${ARROW_RAPIDJSON_REQUIRED_VERSION})
```

Comment:
Does this have any effect?

Reply:
This enables version check by CMake.

https://cmake.org/cmake/help/latest/command/find_package.html

> The [version] argument requests a version with which the package found should be compatible (format is major[.minor[.patch[.tweak]]]). The EXACT option requests that the version be matched exactly. If no [version] and/or component list is given to a recursive invocation inside a find-module, the corresponding arguments are forwarded automatically from the outer call (including the EXACT flag for [version]). Version support is currently provided only on a package-by-package basis (see the Version Selection section below).

This corresponds to `VERSION_VER` of `find_package_handle_standard_args()` in `FindRapidJSONAlt.cmake`.

https://cmake.org/cmake/help/latest/module/FindPackageHandleStandardArgs.html

> VERSION_VAR <version-var>
> 
> Specify the name of a variable that holds the version of the package that has been found. This version will be checked against the (potentially) specified required version given to the find_package() call, including its EXACT option. The default messages include information about the required version and the version which has been actually found, both if the version is ok or not.

CMake checks version when we specify the required version to `find_package()` and pass the detected version to `find_package_handle_standard_args()` in `FindRapidJSONAlt.cmake`.

Reply:
Thank you, I did not know about the version checking logic of `find_package_handle_standard_args`